metrics/docs/02_sign_and_notarize_metrics_binaries.md

```diff
@@ -0,0 +1,176 @@
+- [notarize-cli](https://github.com/bacongravy/notarize-cli)
+- [Import codesign certificate Github Action](https://github.com/Apple-Actions/import-codesign-certs)
+
+# Analyze
+
+- [Requirements](#requirements)
+- [Analyzing process](#analyzing-process)
+- [Decision](#decision)
+
+## Requirements
+
+Before we can use tools, describes below, we need to:
+
+ - get an Apple ID
+ - generate an [app specific password](https://support.apple.com/en-us/HT204397)
+ - get the [Developer ID Application certificate](#developer-id-application)
+
+### Developer ID Application
+
+To get the certificate follow the next steps:
+ 1. On macOS open `Keychain Access` and select the `Keychain Access` button on the top left corner of the screen.
+ 2. Select the `Certificate Assistant` and then `Request a Certificate from a Certificate Authority`.
+ 3. Open the [following link](https://developer.apple.com/).
+ 4. Go to `Certificates IDs & Profiles`.
+ 5. Create a certificate.
+ 6. Choose the `Developer ID Application` from the list.
+ 7. Upload the signing request that we’ve created before.
+ 8. Download the created certificate and add it to your (default keychains) Keychain Access.
+ 9. Export the added certificate with the file format: `Personal information exchange(.p12)`.
+
+Extra step:
+
+To upload the exported certificate to the `Github Secrets`, we need to copy the base64 version:
+
+```
+base64 Certificates.p12 | pbcopy
+```
+
+And then paste the result to the `secrets`.
+
+## Analyzing process
+
+The analysis begins with the selection of programs and the study of their work, complexity, and popularity. This provides a comprehensive view of how to solve the given problem.
+
+The research concludes with a short description of the tool(method), an example of the usage, and a list of pros and cons of each.
+
+### [gon](https://github.com/mitchellh/gon)
+
+`gon` is a simple tool for signing and notarizing CLI binaries for macOS. It can be run manually or in automation pipelines. `gon` requires a configuration file (`.hcl` or `.json`) that can be specified as a file path or passed in via stdin. The configuration specifies all the settings tool will use to sign and package files.
+Before using `gon` you should install it:
+
+```
+brew tap mitchellh/gon
+brew install mitchellh/gon/gon
+```
+
+To sign and notarize CLI binary with `gon` you should run the following command:
+
+```
+gon [flags] [CONFIG]
+```
+
+where `flags` are:
+ - `-log-level` - output logs in JSON format for machine readability
+ - `-log-json` - log level to output (defaults to no logging) 
+
+Pros: 
+ - can be run manually or via pipelines (Github Actions etc.)
+ - single file for configuration (options can be passed via stdin)
+ - a debug can be enabled with `-log-json` flag
+ - can be run with `notarization-only` configuration
+ - notarize packages and wait for the notarization to complete
+ - signed files into a `.dmg` or `.zip`
+ - popular tool
```

Comment:
would be great to provide a little bit of explanation on what popular means - stars in github etc...